crates/api_models/src/events/customer.rs

```diff
@@ -5,31 +5,33 @@ use crate::customers::{CustomerDeleteResponse, CustomerId, CustomerRequest, Cust
 impl ApiEventMetric for CustomerDeleteResponse {
     fn get_api_event_type(&self) -> Option<ApiEventsType> {
         Some(ApiEventsType::Customer {
-            customer_id: self.customer_id.clone(),
+            customer_id: Some(self.customer_id.clone()),
         })
     }
 }
 
 impl ApiEventMetric for CustomerRequest {
     fn get_api_event_type(&self) -> Option<ApiEventsType> {
-        self.customer_id
+        self.get_merchant_reference_id()
             .clone()
-            .map(|customer_id| ApiEventsType::Customer { customer_id })
+            .map(|customer_id| ApiEventsType::Customer {
+                customer_id: Some(customer_id),
+            })
     }
 }
 
 impl ApiEventMetric for CustomerResponse {
     fn get_api_event_type(&self) -> Option<ApiEventsType> {
         Some(ApiEventsType::Customer {
-            customer_id: self.customer_id.clone(),
+            customer_id: self.get_merchant_reference_id().clone(),
```

Comment:
For now lets keep the customer_id as mandatory,
in places where we don't have a customer id we can just not log the event for now by returning a `None`

```rust
impl ApiEventMetric for CustomerResponse {
    fn get_api_event_type(&self) -> Option<ApiEventsType> {
        self.get_merchant_reference_id().clone().map(|cid| 
            ApiEventsType::Customer { customer_id: cid } 
        )
    }
}
```